log4j-core/src/main/java/org/apache/logging/log4j/core/time/internal/SystemClock.java

```diff
@@ -16,13 +16,19 @@
  */
 package org.apache.logging.log4j.core.time.internal;
 
+import java.time.Instant;
+
 import org.apache.logging.log4j.core.time.Clock;
+import org.apache.logging.log4j.core.time.MutableInstant;
+import org.apache.logging.log4j.core.time.PreciseClock;
 
 /**
  * Implementation of the {@code Clock} interface that returns the system time.
+ * @since 2.11.0
  */
+// Precise clock is not implemented because the instant() method in the init method is not garbage free.
```

Comment:
Hmm, this would result in a regression in some places, we use microsecond precision everywhere at this point at work since events can be scrambled within millisecond buckets by our indexer. I suspect we'd be better of implementing PreciseClock and updating our tests, but I have a very specific use case. Either way we should swing back to add another clock implementation to support whichever implementation SystemClock does not.

Reply:
I agree. If it is possible to detect that we are not in "garbage free mode" then we could go ahead with the init method. I admittedly took a shortcut has I spent nearly a week getting this done and I didn't want to solve every problem in Log4j.